libfaad/decoder.c

```diff
@@ -641,16 +640,16 @@ static void create_channel_config(NeAACDecStruct *hDecoder, NeAACDecFrameInfo *h
             hInfo->channel_position[chpos++] = FRONT_CHANNEL_CENTER;
             chdir--;
         }
-        for (i = 0; i < chdir; i += 2)
+        for (i = 0; i < chdir; i++)
         {
-            hInfo->channel_position[chpos++] = FRONT_CHANNEL_LEFT;
-            hInfo->channel_position[chpos++] = FRONT_CHANNEL_RIGHT;
+            hInfo->channel_position[chpos++] =
+                (i & 1) ? FRONT_CHANNEL_RIGHT : FRONT_CHANNEL_LEFT;
```

Comment:
Why this semantic change?

Reply:
Oh wait, I misread the diff!

Reply:
As it is noted above, CPE flag is ignored. Given that some entries are CPE, the number of channels could be odd.
Before the change loop was striding with step 2, and for odd number of channels created one extra entry.
In the new code for even number of channels result remains same as before. For odd number we do not create extra item.

Ideally, we should take CPE flag into account, but I'm not sure how this should be done (without access to spec).
So I decided to fix it with minimal change in code / results.